File: mingachevir-real-estate-agency-server/src/main/java/io/mingachevir/mingachevirrealestateserver/model/request/CreateHouseRequest.java
package io.mingachevir.mingachevirrealestateserver.model.request;

import io.mingachevir.mingachevirrealestateserver.util.PriceType;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

@Getter
@Setter
@Data
public class CreateHouseRequest {
    // for update
    private Long id;
    List<MultipartFile> deletedMultipartFiles;

    //for requester
    private String requester;
    private Integer number;

    private String name;
    private String description;
    private Double price;
    private PriceType priceType;
    private String location;
    private String type;

    private String coverImage;

    private Long mainCategoryId;
    private Long subCategoryId;


    List<String> imagesPaths;

    private List<Long> selectiveParameterValuesIds;

    private List<CreateInputParameterValueRequest> inputParameterValues;
}

File: mingachevir-real-estate-site-master/src/app/models/request/CreateHouseRequest.ts
import {CreateInputParameterValueRequest} from "./CreateInputParameterValueRequest";

export interface CreateHouseRequest {
  id?: number | null;
  deletedMultipartFiles?: File[];

  requester: string;
  number: number;

  name: string | null;
  description: string | null;
  price: number | null | undefined;
  priceType: string | null | undefined;
  location: string | null | undefined;
  type: string | null;

  mainCategoryId: number | null | undefined;
  subCategoryId: number | null | undefined;

  imagesPaths?: string[] | [];
  coverImage?: string | null;

  selectiveParameterValuesIds: number[];
  inputParameterValues: CreateInputParameterValueRequest[];
}
